test/state/host.cpp

```diff
@@ -74,24 +74,38 @@ uint256be Host::get_balance(const address& addr) const noexcept
     return (acc != nullptr) ? intx::be::store<uint256be>(acc->balance) : uint256be{};
 }
 
+namespace
+{
+// For EXTCODE* instructions if the target is an EOF account, then only return EF00.
+// While we only do this if the caller is legacy, it is not a problem doing this
+// unconditionally, because EOF contracts dot no have EXTCODE* instructions.
+bytes_view extcode(bytes_view code) noexcept
+{
+    if (code.size() >= 2 && code[0] == 0xEF && code[1] == 0x00)
+        return code.substr(2);
+
+    return code;
+}
+}  // namespace
+
 size_t Host::get_code_size(const address& addr) const noexcept
 {
     const auto* const acc = m_state.find(addr);
-    return (acc != nullptr) ? acc->code.size() : 0;
+    return (acc != nullptr) ? extcode(acc->code).size() : 0;
 }
 
 bytes32 Host::get_code_hash(const address& addr) const noexcept
 {
     // TODO: Cache code hash. It will be needed also to compute the MPT hash.
     const auto* const acc = m_state.find(addr);
-    return (acc != nullptr && !acc->is_empty()) ? keccak256(acc->code) : bytes32{};
+    return (acc != nullptr && !acc->is_empty()) ? keccak256(extcode(acc->code)) : bytes32{};
```

Comment:
Patching `EXTCODEHASH` will be a headache for real implementations. The code hash is available directly in the account node. Now you need to additionally load the code to obtain a possibly different hash.

Reply:
> Now you need to additionally load the code to obtain a possibly different hash.

This was always the case with this approach. Need to figure out if it is an EOF account or not (by loading the account).

Reply:
`keccak256("ef00")` should better be memoized or hard-coded

Reply:
I think this is generally fine for now although it conflicts with the TODO comment above.